src/Form/Expression/ContextFormTrait.php

```diff
@@ -58,6 +58,8 @@ public function buildContextForm(array $form, FormStateInterface $form_state, $c
     $element = &$form['context'][$context_name]['setting'];
 
     if ($mode == 'selector') {
+      // Always use a textfield for selector mode.
+      $element['#type'] = 'textfield';
       $element['#description'] = $this->t("The data selector helps you drill down into the data available to Rules. <em>To make entity fields appear in the data selector, you may have to use the condition 'entity has field' (or 'content is of type').</em> More useful tips about data selection is available in <a href=':url'>the online documentation</a>.", [
         ':url' => 'https://www.drupal.org/node/1300042',
```

Comment:
This is so that the element goes back to a textfield if the data selector is used.